Gameplay/Ability.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Ability.h"
+#include "..//CSC8503/CSC8503Common/GameObjectGenerator.h"
+#include <string>
+#include <iostream>
+
+Ability::Ability(const char* colourName, int abilityNumber) {
+	std::ifstream file("Ability.json");
+	std::stringstream buffer;
+	buffer << file.rdbuf();
+
+	rapidjson::Document document;
+	document.Parse(buffer.str().c_str());
+
+	if (document.HasMember(colourName))
+	{
+		rapidjson::Value& objects = document[colourName];
+		
+		cooldown = objects[abilityNumber]["cooldown"].GetFloat();
+		range = objects[abilityNumber]["range"].GetFloat();
+		GetVector(objects[abilityNumber], "size", size);
+		startUp = objects[abilityNumber]["cooldown"].GetFloat();
+		endLag = objects[abilityNumber]["cooldown"].GetFloat();
+	}
```

Comment:
Could we not load the colour properties into memory, instead of having to read the file in game time

Reply:
I was thinking this would fulfil a "Data Driven" better. Even if its a bit extra.

Reply:
You can still make it data driven, just load the json into a container class, and then read from that container class instead of the json directly

Reply:
Ok, I'll have a stab at it.